### DISCUSSION TITLE
How to use the Processes in the generic tech pdk?

### DISCUSSION THREAD

**Discussion by duarte-jfs** [gdsfactory/gdsfactory#2856] [Q&A] [answered]
I am writing my own pdk using gdsfactory for the first  time and I am puzzled on how to use the `gf.technology.processes` module. Can anyone give an example on how they would be useful?

**Comment by joamatab** [comment]
yes, maybe we should add some docs about it, this is useful for process simulation and TCAD

@simbilod

**Reply by duarte-jfs** [reply]
In what way could it be useful for process simulation?

**Reply by joamatab** [reply]
for TCAD sims

**Comment by simbilod** [comment, marked as answer]
> I am writing my own pdk using gdsfactory for the first  time and I am puzzled on how to use the `gf.technology.processes` module. Can anyone give an example on how they would be useful?

They are used in the Sentaurus plugin, which is not and will not be documented, but is present in the repo.

In general they are just data classes that are convenient to hold process step information. That way a plugin can take a Component + process steps and setup process simulation.

**Reply by duarte-jfs** [reply]
I see. That makes sense. Initially I thought it was for pyxs, but then I saw another module for pyxs which doesn't use this so I was confused. But that's good to know. Thank you!

**Comment by simbilod** [comment]
In the future we would like to tie it to the Layer stack, for instance generate a LayerStack from the list of process steps
